FirestoreLINQ/Extensions.cs

```diff
@@ -9,5 +11,18 @@ public static IQueryable<T> AsQuerable<T>(this CollectionReference collection)
             IQueryable<T> source = new Queryable<T>(collection);
             return source;
         }
+
+        public static CollectionReference Collection<T>(this FirestoreDb db) where T : class
+        {
+            TypeInfo typeInfo = typeof(T).GetTypeInfo();
+            FirestoreCollectionAttribute collectionAttribute = FirestoreCollectionAttribute.GetAttributes(typeInfo);
```

Comment:
The process you followed to get the collection name is a very long process.
Please follow this process:
`var attrib = FirestoreCollectionAttribute(Attribute.GetCustomAttributes(property).SingleOrDefault(x => x is FirestoreCollectionAttribute));`

Reply:
And also throw an exception if the collection name is empty or the attribute is not configured.